File: microservices/energydevices/src/main/java/com/cezar/energydevices/DTO/DeviceDTO.java
package com.cezar.energydevices.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class DeviceDTO {
    private String description;
    private String address;
    private Float maxHourlyConsumption;
    private Long ownerId;
}

File: energy_frontend/src/app/DTO/DeviceDTO.ts
export class DeviceDTO {
    public id : number;
    public description : string;
    public address : string;
    public maxHourlyConsumption : number;
    public owner : number;

    constructor(id : number, description : string, address : string, maxHourlyConsumption : number, owner : number){
        this.id = id;
        this.description = description;
        this. address = address;
        this.maxHourlyConsumption = maxHourlyConsumption;
        this.owner = owner;
    }
}
  